pandas/_libs/hashtable.pyx

```diff
@@ -55,6 +55,26 @@ cdef Py_ssize_t _INIT_VEC_CAP = 128
 include "hashtable_class_helper.pxi"
 include "hashtable_func_helper.pxi"
 
+
+# map derived hash-map types onto basic hash-map types:
+if np.dtype(np.intp) == np.dtype(np.int64):
```

Comment:
any idea of this check can be done at compile-time?

Reply:
There are at least two options:

Using cython's conditionals statements( https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#conditional-statements), e.g.

```
IF UNAME_MACHINE=='x86_64':
    # do 64bit stuff here
ELSE:
    # do 32bit stuff here
```

I used it sometimes for Linux/Windows but never for 64bit/32bit, so not sure there are some pitfalls.

Another one would be to use tempita, which is kind of closer to the current code:

```
{{py:

import numpy as np

}}

{{if np.dtype(np.int64) == np.dtype(np.intp)}}
# 64bit stuff
{{elif np.dtype(np.int32) == np.dtype(np.intp)}}
# 32bit stuff
{{else}}
XXX=YYYY # cannot be compiled
{{endif}}
```

What I ask myself: Why bother? The code is executed only once when the module is loaded, it will never fail on plattforms supported by pandas (the else-branch will never run)...

Reply:
> What I ask myself: Why bother? The code is executed only once when the module is loaded, it will never fail on plattforms supported by pandas (the else-branch will never run)...

Fair enough.  You've thought this through plenty, and I trust your judgement on this.

Reply:
Another option would be to use implement a fused type wrapper for the lower-level kh_foo functions, then within unique_label_indices just call eg `kh_append` instead of `kh_append_int64`.

This would also let us de-tempita some of the other hashtable code (which weve been aiming to do more generally).  But last time I tried doing this performance suffered for reasons I never figured out.

Reply:
> Using cython's conditionals statements( https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#conditional-statements), e.g.

i asked on the cython mailing list if there is a robust way to do this and got a resounding "no": https://groups.google.com/g/cython-users/c/pGxvx22b6QQ